release/models/platform/openconfig-platform-common.yang

```diff
@@ -232,9 +238,12 @@ submodule openconfig-platform-common {
       type oc-platform-types:component-power-type;
       default POWER_ENABLED;
       description
-        "Enable or disable power to the component";
+        "When set to POWER_DISABLED, the component should be shut down by removing
+        electrical power.  This is intended to be used to prevent the component
+        from becoming active even after a reboot of the system.  A component may
+        not honor power-admin-state depending on rules defined in the description
+        of the component config container.";
```

Comment:
I suggest if a component does not support this leaf, the device should return an error.

Reply:
This basically considers that the implementation does support the leaf. However, if the component is a controller-card and if it happens to be PRIMARY, then it will take the configuration of POWER_DISABLED for "config/power-admin-state" but it will not power-off because it is the PRIMARY. So in this case "state/power-admin-state" will still be "POWER_ENABLED"